components/application-configuration/api/src/main/java/org/phenotips/configuration/internal/consent/DefaultConsentAuthorizer.java

```diff
@@ -0,0 +1,152 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses/
+ */
+package org.phenotips.configuration.internal.consent;
+
+import org.phenotips.configuration.RecordElement;
+import org.phenotips.data.Consent;
+import org.phenotips.data.ConsentManager;
+import org.phenotips.data.ConsentStatus;
+import org.phenotips.data.Patient;
+
+import org.xwiki.component.annotation.Component;
+
+import java.util.LinkedList;
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import org.apache.commons.codec.binary.StringUtils;
+
+/**
+ * The default implementation of a {@link ConsentAuthorizer}. This is a temporary implementation, as the whole
+ * application-configuration section must be redesigned.
+ *
+ * @version $Id$
+ * @since 1.2RC2
+ */
+@Component
+@Singleton
+public class DefaultConsentAuthorizer implements ConsentAuthorizer
+{
+    @Inject
+    private ConsentManager consentManager;
+
+    @Override public boolean consentsGloballyEnabled()
+    {
+        return !consentManager.getSystemConsents().isEmpty();
+    }
+
+    @Override
+    public List<RecordElement> filterForm(List<RecordElement> elements, Patient patient)
+    {
+        if (!this.consentsGloballyEnabled()) {
+            return elements;
+        }
+        List<Consent> consents = this.consentManager.loadConsentsFromPatient(patient);
+        if (!authorizeInteraction(consents)) {
+            return new LinkedList<>();
+        }
+
+        List<String> granted = getGrantedIds(consents);
+        List<RecordElement> updatedElements = new LinkedList<>();
+        for (RecordElement element : elements) {
+            if (this.isElementEnabled(element, granted)) {
+                updatedElements.add(element);
+            }
+        }
+        return updatedElements;
+    }
+
+    @Override
+    public boolean authorizeInteraction(Patient patient)
+    {
+        List<Consent> consents = this.consentManager.loadConsentsFromPatient(patient);
+        return this.authorizeInteraction(consents);
+    }
+
+    @Override public boolean authorizeInteraction(Iterable<String> grantedConsents)
+    {
+        for (Consent consent : this.consentManager.getSystemConsents()) {
+            boolean found = false;
+            if (consent.isRequired()) {
+                for (String granted : grantedConsents) {
+                    if (StringUtils.equals(granted, consent.getId())) {
+                        found = true;
+                        break;
+                    }
+                }
+                if (!found) {
+                    return false;
+                }
+            }
+        }
+        return true;
+    }
```

Comment:
:(

You can still clean it up, since you're essentially re-implementing `Collection.contains()`.

```java
boolean found = true;

Collection<String> grantedConsentsCollection = IteratorUtils.toList(grantedConsents.iterator());

for (Consent consent : this.consentManager.getSystemConsents()) {
    if (!consent.isRequired()) {
        continue;
    }

    found = grantedConsentsCollection.contains(consent.getId());

    if (!found) {
        break;
    }
}

return found;
```